### PR TITLE
Add bazel_codegen to this repository.

### PR DESCRIPTION
**Pull request by matanlurey**
Partial work towards https://github.com/dart-lang/build/issues/931.

... this is a 1:1 copy, with minimum edits in order to comply with `implicit-casts: false`, of which some (not all) of these would be Dart2 failures anyway. I opened up https://github.com/dart-lang/args/issues/95 to try and find nicer ways of doing some of these casts.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,145 +1,148 @@
 # Created with https://github.com/dart-lang/mono_repo
 language: dart
 
-# Hand-edited
```
Comment (jakemac53) [inline .travis.yml]: Need to restore this one... seem familiar? lol

File: .travis.yml [hunk 2]
```diff
@@ -1,145 +1,148 @@
...
+      dart: dev
+    - stage: unit_test
+      script: ./tool/travis.sh test_0
+      env: PKG="bazel_codegen"
+      dart: dev
     - stage: analyze_and_format
       script: ./tool/travis.sh dartfmt dartanalyzer
       env: PKG="build"
-      dart: 2.0.0-dev.28.0
+      dart: dev
```
[the remark this answers is not in the document]
Comment (natebosch) [reply .travis.yml]: If Travis isn't passing on latest and we need to go back to dev.28 we'll also have to revert the formatting changes.

File: .travis.yml (hunk 1 above)
Comment (matanlurey) [reply .travis.yml]: Added back.

File: .travis.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (matanlurey) [reply .travis.yml]: Ultimately we'll need to use the latest anyway... Let me keep trying.

**Comment by matanlurey** [comment]
Looks like this will timeout still...

`05:49 +2 ~1: compiling test/hello_world_test.dart                              `

... similar to the issue for AngularDart and others :-/

/cc @grouma

**Comment by matanlurey** [comment]
Green! PTAL @natebosch @jakemac53

**Comment by jakemac53** [comment]
@matanlurey are you willing to handle also deleting this from the original repo?

**Comment by matanlurey** [comment]
@jakemac53 Yup! I'll send a PR out after you merge :)

**Pull request merged**